src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java

```diff
@@ -577,6 +580,54 @@ public List<Movement> getMovementForPrint(
 		}
 		return pMovement;
 	}
+	
+	/**
+	 * Retrieves all the lot referred to the specified {@link Medical}, expiring first on top
+	 * 
+	 * @param medical the medical.
+	 * @return a list of {@link Lot}.
+	 * @throws OHServiceException if an error occurs retrieving the lot list.
+	 */
+	public List<Lot> getAllLotsByMedical(Medical medical) throws OHServiceException {
```

Comment:
Why to repeat a full method with only one line of code different? use overloading and DRY principle

Reply:
I have removed it and update method `getLotsByMedical(Medical medical)` by this `getLotsByMedical(Medical medical, boolean removeEmpty)`